### PR TITLE
Feature/templateengine First version of template engine for iofog franckOL#1

### PR DESCRIPTION
**Pull request by franckOL**
Signed-off-by: Roudet Franck <franck.roudet@orange.com>

- Does the iofog.org or README.md documentation need to change?
  - [ ] Yes 
  - [ ] No

### Description
Early stage development for template engine based on liquidjs. see franckOL#1 .

The first version has only a self object to help configuration, and it refere to the document itself.
The template engine liquidjs is used for all POST, PUT verbs for all API routes.

Next improvements:
- [ ] Only apply on precise route: routes use by agent don't need that
- [ ] Add all controller applications to context of template rendering (lazy loading)
- [ ] Add all controller edge ressources to context of template rendering (lazy loading)
- [ ] Add all controller agent to context of template rendering (lazy loading)

### CODE REVIEW

**Review by Pixcell — CHANGES_REQUESTED** [review]
Looks great!
Im just waiting for the postman_tests

**Comment by franckOL** [comment]
> Looks great!
> Im just waiting for the postman_tests

I thing it's OK now

File: src/helpers/template-helper.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  * Object in depth traversal and right value templateEngine rendering
+  * @param {*} subjects
+  * @param {*} templateContext
+  */
+const rvaluesVarSubstition = async (subjects, templateContext) => {
+  Object.keys(subjects).forEach((key) => {
+    if (typeof subjects[key] === 'object') {
+      rvaluesVarSubstition(subjects[key], templateContext)
+    } else if (typeof subjects[key] === 'string') {
+      subjects[key] = templateEngine.parseAndRenderSync(subjects[key], templateContext)
```
Comment (baghbidi) [inline src/helpers/template-helper.js:27]: Does `parseAndRenderSync` handle circular references? Like what would happen if someone submits this template:
```
apiVersion: iofog.org/v2
kind: Application
metadata:
  name: "{{ self.metadata.name  }}"
spec:
...
```

**Review by baghbidi — APPROVED** [review]
Left a comment, otherwise looks good to me.

File: src/helpers/template-helper.js (hunk 1 above)
Comment (franckOL) [reply src/helpers/template-helper.js:27]: not circular in fact: the value of the key is going to be replace by itself, and no more variable expansion is going to be done further so the result is:`{{ self.metadata.name  }}`

**Pull request merged**